src/snapred/backend/recipe/FetchGroceriesRecipe.py

```diff
@@ -0,0 +1,152 @@
+        else:
+            return f"{runConfig.IPTS}nexus/SNAP_{runConfig.runNumber}.nxs.h5"
+
+    def _createNexusWorkspaceName(self, runConfig) -> str:
+        name: str = f"_TOF_RAW_{runConfig.runNumber}"
+        if runConfig.isLite:
+            name = name + "_lite"
+        return name
+
+    def fetchNexusData(self, runConfig: RunConfig, loader: str = "") -> Dict[str, Any]:
+        """
+        Fetch just a nexus data file
+        inputs:
+        - runConfig, a RunConfig object corresponf to the data desired
+        - loader, the loading algorithm to use, if you think you already know
+        outputsa dictionary with
+        - "result": true if everything ran correctly
+        - "loader": the loader that was used by the algorithm, use it next time
+        - "workspace": the name of the workspace created in the ADS
+        """
+
+        workspaceName = self._createNexusWorkspaceName(runConfig)
+        fileName = self._createFilenameFromRunConfig(runConfig)
+        data: Dict[str, Any] = {
+            "result": False,
+            "loader": loader,
+            "workspace": workspaceName,
+        }
+        thisKey = (runConfig.runNumber, runConfig.IPTS, runConfig.isLite)
+        if self._loadedRuns.get(thisKey, 0) > 0:
+            newWorkspaceName = f"{workspaceName}_{self._loadedRuns[thisKey]}"
+            self._loadedRuns[thisKey] += 1
+            logger.info("Data already loaded... continuing")
+            CloneWorkspace(
+                InputWorkspace=workspaceName,
+                OutputWorkspace=newWorkspaceName,
+            )
+            data["result"] = True
+            data["workspace"] = newWorkspaceName
+        else:
+            logger.info(f"Fetching nexus data for run {runConfig.runNumber} at {fileName}")
+            algo = FetchAlgo()
+            algo.initialize()
+            algo.setProperty("Filename", fileName)
+            algo.setProperty("Workspace", workspaceName)
+            algo.setProperty("LoaderType", loader)
+            try:
+                data["result"] = algo.execute()
+                data["loader"] = algo.getPropertyValue("LoaderType")
+            except RuntimeError as e:
+                raise RuntimeError(str(e).split("\n")[0]) from e
+            logger.info("Finished loading nexus data")
+            self._loadedRuns[thisKey] = 1
+        return data
+
+    def _createGroupingFilename(self, groupingScheme: str, isLite: bool = True):
+        location = "/SNS/SNAP/shared/Calibration/Powder/PixelGroupingDefinitions/"
+        if isLite:
+            return f"{location}SNAPFocGroup_{groupingScheme}.lite.nxs"
+        else:
+            return f"{location}SNAPFocGroup_{groupingScheme}.xml"
+
+    def _createGroupingWorkspaceName(self, groupingScheme: str, isLite: bool = True):
+        if isLite:
+            return f"SNAPLite_grouping_{groupingScheme}"
+        else:
+            return f"SNAP_grouping_{groupingScheme}"
```

Comment:
these should also be present in the application.yml somehow

Reply:
I think I got this, lemme know if I should move it